openssl/src/x509/mod.rs

```diff
@@ -838,10 +847,21 @@ impl<'a> Iterator for X509NameEntries<'a> {
 
     fn next(&mut self) -> Option<&'a X509NameEntryRef> {
         unsafe {
-            self.loc =
-                ffi::X509_NAME_get_index_by_NID(self.name.as_ptr(), self.nid.as_raw(), self.loc);
+            let entry_count = ffi::X509_NAME_entry_count(self.name.as_ptr());
```

Comment:
Can you move the None checks into the individual branches of the match below since they're each looking for a different thing?